### ISSUE TITLE
Ag-solo wallet slow to deploy: 25 round-trips

### ISSUE THREAD

**Issue by dckc** [Agoric/agoric-sdk#3802]
## Description

I provisioned a testnet client as usual using docker and the testnet faucet.

https://testnet.explorer.agoric.net/account/agoric13lk0cnmehvfqtpcynwq7hx6qu86qtw3cu00qxe

Getting the wallet fully deployed took 22 minutes:

```
2021-08-24T15:39:35.513Z chain-cosmos-sdk: delivering to chain (trips=1)
...
2021-08-24T15:51:51.716Z chain-cosmos-sdk: delivering to chain (trips=20)
...
2021-08-24T16:01:27.652Z chain-cosmos-sdk: delivering to chain (trips=32)
```

## Expected behavior

Didn't this used to take more like 5 minutes?

_p.s. The 22 minutes have gone and we're back to around 5 minutes, but we still take 25 round trips to the chain and back. Is that acceptable?_

## Desktop
 - OS: Ubuntu linux 18.04; docker
 - Browser: chrome

```
$ ag-cosmos-helper version --long
name: agoriccosmos
server_name: ag-cosmos-helper
version: 0.26.9
commit: 08ea8bbc8
build_tags: ledger,
go: go version go1.15.8 linux/amd64
...
```

cc @warner @michaelfig @rowgraus @katelynsills

**Comment by dckc** [comment]
This was noted as a Phase 1 priority in [today's community call](https://agoric.com/event/agoric-community-call-12/).

**Comment by dckc** [comment]
@mhofman would you please surface the wallet install time in the daily perf stuff? That seems like the next step on this issue. (Feel free to split out that part as a separate issue if you like).

I see it on the testnet retrospective under "next steps".

p.s. Another next step that @dtribble suggested is to look at the wallet code to see why it takes 32 round trips. @rowgraus I leave it to you to nominate somebody for that part.

**Comment by dckc** [comment]
@michaelfig the sequence diagram we discussed is in...

 - [agoric start performance investigation](https://gist.github.com/dckc/e29a939208172f16b79ef5f31e6e7d5d)

### Findings

- `onBind` look like they could go in parallel
- `E(zoe).startInstance()` could take an `ERef<Instance>` (see also: #3837)
- wallet API is polling timer

**Comment by mhofman** [comment]
The [loadgen now captures](https://github.com/Agoric/testnet-load-generator/pull/38) the wallet deployment time and is one of the metric included in the spreadsheet based stats. We actually noticed a regression recently thanks to it.

We could still look at opportunities to pipeline the wallet more, but that should be less pressing once we have #3995. Removing myself and assigning @michaelfig

**Comment by Tartuffo** [comment]
Currently takes about 5 minutes to fully deploy.  Closing this as it should be covered by loadgen work.

**Comment by dckc** [comment]
Adding **technical-debt** as I pay significant interest on it most days.

**Comment by abebeos** [comment]
@dckc , my understanding here is

* original issue "Getting the wallet fully deployed took 22 minutes:" is solved
* a follow-up issue "solo wallet takes 25 round-trips to deploy" came up

It is sometimes better to just close the fixed issue, and file a follow-up issue with the relevant information and a backpointer (like "spawned by #3802"). A capable newcomer (knowledge in general IT/programming, but not much knowledge of Agoric) should be able to look into the issue without much effort.


## solo wallet takes 25 round-trips to deploy

(spawned by: #3802)

(explain round-trips)
(explain reproduction and how to measure the round-trips)

**Comment by dckc** [comment]
> It is sometimes better to just close the fixed issue ...

perhaps. In my estimation, not this time.

**Comment by abebeos** [comment]
> perhaps. In my estimation, not this time.

Very well. I extract that this issue (like several others) are addressing mostly the core-team itself, and are not meant for interested contributors.
